core/src/main/java/com/linecorp/armeria/client/endpoint/AbstractListenable.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.function.Consumer;
+
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * A skeletal {@link Listenable} implementation.
+ */
+abstract class AbstractListenable<T> implements Listenable<T> {
```

Comment:
`Listenable` and `AbstractListenable` could be moved to `com.linecorp.armeria.common.util`?